utils/rsaencryption/rsa_encryption_test.go

```diff
@@ -46,3 +55,56 @@ func TestPrivateKeyToByte(t *testing.T) {
 	require.NotNil(t, b)
 	require.Greater(t, len(b), 1024)
 }
+
+func TestConvertEncryptedPemToPrivateKey(t *testing.T) {
+	keystorePassword := "123123123"
+	generatedPrivateKey, err := rsa.GenerateKey(rand.Reader, keySize)
+	require.NoError(t, err)
+	privDER := x509.MarshalPKCS1PrivateKey(generatedPrivateKey)
+
+	// Create a pem.Block with the private key.
+	privateBlock := pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: privDER,
+	}
+
+	// Encode the pem.Block with the private key into a temporary file.
+	tmpPrivateFile, err := os.CreateTemp("", "private_key.pem")
+	require.NoError(t, err)
+	err = pem.Encode(tmpPrivateFile, &privateBlock)
+	require.NoError(t, err)
+	err = tmpPrivateFile.Close()
+	require.NoError(t, err)
+
+	// Encrypt the private key file using OpenSSL.
+	passString := "pass:" + keystorePassword
+	cmd := exec.Command("openssl", "rsa", "-aes256", "-in", tmpPrivateFile.Name(), "-out", "encrypted_private_key.pem", "-passout", passString)
+	err = cmd.Run()
+	require.NoError(t, err)
+
+	// Delete the temporary unencrypted private key file.
+	err = os.Remove(tmpPrivateFile.Name())
```

Comment:
If `openssl` fails, the file won't be removed. Please remove it in `defer` after `os.CreateTemp` call

Reply:
Example: https://github.com/golang/go/blob/13529cc5f443ef4e242da3716daa0032aa8d34f2/src/net/error_test.go#L734